cider/exec/module/CiderCompileModule.cpp

```diff
@@ -86,10 +86,15 @@ CiderTableSchema CiderCompilationResult::getOutputCiderTableSchema() const {
   return impl_->getOutputCiderTableSchema();
 }
 
+QueryType CiderCompilationResult::getQueryType() const {
+  return impl_->getQueryType();
+}
+
 class CiderCompileModule::Impl {
  public:
   Impl() {
-    executor_ = Executor::getExecutor(Executor::UNITARY_EXECUTOR_ID, nullptr, nullptr);
+    id_ = std::hash<std::thread::id>{}(std::this_thread::get_id());
+    executor_ = Executor::getExecutor(id_, nullptr, nullptr);
```

Comment:
why not make executors_ non static, then owned by CiderRuntimeModule?

Reply:
`executor_` is non-static, and it will call code gen path, actually not used in `CiderRuntimeModule`

Reply:
getExecutor find one executor from `std::map<int, std::shared_ptr<Executor>> Executor::executors_;` which is static and global. we should change this to non static, then the executor_ will be local to the CiderCompileModule( CiderRuntimeModule was a typo)

Reply:
got you. Maybe we can do this refactor later. I am not sure whether this will bring extra issues.

Reply:
LGTM